Advanced/Play Asset Delivery/Assets/PlayAssetDelivery/Runtime/InitializeAddressables.cs

```diff
@@ -0,0 +1,118 @@
+#endif
+        }
+
+        void Setup()
+        {
+            Addressables.ResourceManager.ResourceProviders.Add(new PlayAssetDeliveryAssetBundleProvider());
+            Addressables.ResourceManager.InternalIdTransformFunc = AssetPackTransformFunc;
+            hasInitialized = true;
+        }
+
+        void CheckDownloadStatus(AndroidAssetPackInfo info)
+        {
+            if (info.status == AndroidAssetPackStatus.Failed)
+                Debug.LogError($"Failed to retrieve the state of asset pack '{info.name}'."); 
+            else if (info.status == AndroidAssetPackStatus.Unknown)
+                Debug.LogError($"Asset pack '{info.name}' is unavailable for this application. This can occur if the app was not installed through Google Play.");
+            else if (info.status == AndroidAssetPackStatus.Canceled)
+                Debug.LogError($"Cancelled asset pack download request '{info.name}'.");
+            else if (info.status == AndroidAssetPackStatus.WaitingForWifi)
+            {
+                AndroidAssetPacks.RequestToUseMobileDataAsync(result =>
+                {
+                    if (!result.allowed)
+                        Debug.LogError("Request to use mobile data was denied.");
+                });
+            }
+            else if(info.status == AndroidAssetPackStatus.Completed)
+            {
+                string assetPackPath = AndroidAssetPacks.GetAssetPackPath(info.name);
+                if (string.IsNullOrEmpty(assetPackPath))
+                {
+                    Debug.LogError($"Downloaded asset pack '{info.name}' but cannot locate it on device.");
+                }
+            }
+        }
+    
+        string AssetPackTransformFunc(IResourceLocation location)
+        {
+
+#if UNITY_ANDROID && !UNITY_EDITOR
+            string filename = Path.GetFileName(location.InternalId);
+
+            if (location.ResourceType == typeof(IAssetBundleResource))
+            {
+                string assetPackName = Path.GetFileNameWithoutExtension(location.InternalId);
+                if (PlayerPrefs.HasKey(assetPackName))
+                {
+                    // Load bundle from a custom asset pack. PlayAssetDeliveryBundleProvider.Provider previously stored its path.
+                    return PlayerPrefs.GetString(assetPackName);
+                }
+            }
+
+            string streamingAssetsPackPath = AndroidAssetPacks.GetAssetPackPath("UnityStreamingAssetsPack");
+            if (!string.IsNullOrEmpty(streamingAssetsPackPath))
+            {
+                // Load resource from a fast-follow or on-demand 'UnityStreamingAssetsPack'.
+                return Path.Combine(streamingAssetsPackPath, filename);
+            }
+            
+            // Load resource that was assigned to an install-time core Unity asset pack. 
+            // The pack's contents should be installed in the 'aa/Android' folder in the APK's StreamingAssets path.
+            return location.InternalId;
```

Comment:
If I understand correctly, as I explained in the PlayAssetDeliveryAssetBundleProvider.cs script, this is depending on the incidental behavior which is not officially supported.